openedx/features/smart_referral/helpers.py

```diff
@@ -0,0 +1,57 @@
+from django.contrib.auth.models import User
+
+from lms.djangoapps.onboarding.models import UserExtendedProfile, Organization
+
+CONTACT_EMAIL_KEY = 'contact_email'
+
+
+def get_platform_contacts_and_non_platform_contacts(contacts):
+    """
+    Returns two(2) list first list belongs to contacts of users that are currently registered on our platform and
+    second list of those contacts that are not registered.
+    """
+    platform_users = []
+    non_platform_users = []
+    for contact in contacts:
+        try:
+            User.objects.get(email=contact[CONTACT_EMAIL_KEY])
+            platform_users.append(contact)
+        except User.DoesNotExist:
+            non_platform_users.append(contact)
+    return platform_users, non_platform_users
+
+
+def sort_contacts_by_org_and_user_domain(contact_list,
+                                         first_criteria_org_email_domain, second_criteria_user_email_domain):
+    """
+    Returns sorted list of contacts based on 2(two) criteria first is the organization admin email domain and
+    second is user email domain.
+    """
+    return sorted(contact_list,
+                  key=lambda contact: (
+                      get_email_domain(contact[CONTACT_EMAIL_KEY]) == first_criteria_org_email_domain,
+                      get_email_domain(contact[CONTACT_EMAIL_KEY]) == second_criteria_user_email_domain),
+                  reverse=True)
+
+
+def get_org_admin_email(user):
+    """
+    Return admin email of that organization with which requested user is affiliated.
+    """
+    # org_admin_email_error = None
+    try:
+        user_extended_profile = UserExtendedProfile.objects.get(user_id=user.id)
+        if user_extended_profile.organization:
+            org = Organization.objects.get(id=user_extended_profile.organization.id)
+            return org.admin.email if org.admin else None
+    except UserExtendedProfile.DoesNotExist:
+        pass
```

Comment:
inconsistent return values. Not all paths are returning values.

Reply:
@monofal We will have to return 'None' if we have to return anything from other flows. How about that?

Reply:
Done